huaweicloud/services/waf/resource_huaweicloud_waf_policy.go

```diff
@@ -0,0 +1,230 @@
+)
+
+func ResourceWafPolicyV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWafPolicyV1Create,
+		Read:   resourceWafPolicyV1Read,
+		Update: resourceWafPolicyV1Update,
+		Delete: resourceWafPolicyV1Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"protection_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "log",
+				ValidateFunc: validation.StringInSlice([]string{
+					"log", "block",
+				}, false),
+			},
+			"level": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				ValidateFunc: validation.IntBetween(0, 3),
+			},
+			"options": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"webattack": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"common": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_engine": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_scanner": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_script": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_other": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"webshell": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"cc": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"custom": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"whiteblackip": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"privacy": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"ignore": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"antitamper": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"full_detection": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceWafPolicyV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	wafClient, err := config.WafV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud WAF client: %s", err)
+	}
+
+	createOpts := policies.CreateOpts{
+		Name: d.Get("name").(string),
+	}
```

Comment:
seems we do nothing about *level* and *protection_mode* parameters.

Reply:
added